plugins/catalog-backend-module-ldap/src/ldap/client.ts

```diff
@@ -147,39 +147,57 @@ export class LdapClient {
   async searchStreaming(
     dn: string,
     options: SearchOptions,
-    f: (entry: SearchEntry) => void,
+    f: (entry: SearchEntry) => Promise<void> | void,
   ): Promise<void> {
     try {
+      let awaitList: Array<Promise<void> | void> = [];
```

Comment:
could move this line all the way down to line 168 (after the `if (err)` check) so it's more within its context